vignettes/flowmaps-static.qmd

```diff
@@ -0,0 +1,774 @@
+  st_centroid() |> 
+  st_transform(crs = st_crs(spain_for_join))
+
+ca_distr <- district_centroids |>
+  st_join(spain_for_join) |> 
+  st_drop_geometry() |>
+  filter(!is.na(ccaa.shortname.en)) |> 
+  select(id, ca_name = ccaa.shortname.en)
+
+ca_distr
+```
+
+
+```
+# A tibble: 3,784 × 2
+   id       ca_name       
+   <chr>    <chr>         
+ 1 01001    Basque Country
+ 2 01002    Basque Country
+ 3 01004_AM Basque Country
+ 4 01009_AM Basque Country
+ 5 01010    Basque Country
+ 6 01017_AM Basque Country
+ 7 01028_AM Basque Country
+ 8 01036    Basque Country
+ 9 01043    Basque Country
+10 01047_AM Basque Country
+# ℹ 3,774 more rows
+# ℹ Use `print(n = ...)` to see more rows
+```
+
+This way we get a table with `districts` `id`s and their corresponding autonomous community names.
+
+### Count flows between pairs of autonomous communities
+
+We can now add these ids to the total flows by `districts` `id` pairs and calculate total flows between autonomous communities:
+
+```{r}
+flows_by_ca <- flows_by_district |>
+  left_join(ca_distr |>
+    rename(id_orig = ca_name),
+      by = c("id_origin" = "id")
+    ) |> 
+  left_join(ca_distr |>
+    rename(id_dest = ca_name),
+      by = c("id_destination" = "id")
+    ) |> 
+  group_by(id_orig, id_dest) |>
+  summarise(n_trips = sum(n_trips, na.rm = TRUE),
+    .groups = "drop") |> 
+  rename(id_origin = id_orig, id_destination = id_dest)
+
+flows_by_ca
+```
+
+```
+# A tibble: 349 × 3
+   id_origin id_destination      n_trips
+   <chr>     <chr>                 <dbl>
+ 1 Andalusia Andalusia         21991628.
+ 2 Andalusia Aragon                1405.
+ 3 Andalusia Asturias               777.
+ 4 Andalusia Balearic Islands      1568.
+ 5 Andalusia Basque Country         970.
+ 6 Andalusia Canary Islands        1853.
+ 7 Andalusia Cantabria              268.
+ 8 Andalusia Castile and León      3025.
+ 9 Andalusia Castile-La Mancha    11374.
+10 Andalusia Catalonia             5679.
+# ℹ 339 more rows
+# ℹ Use `print(n = ...)` to see more rows
+```
+
+## Reshaping the data for visualization
+
+We are going to use the `{flowmapper}` [@flowmapper-r] package to plot the flows. This package expects the data to be in the following format:
+
+```
+'data.frame':	7 obs. of  8 variables:
+ $ id_a   : chr  "X1" "X2" "X3" "X3" ...
+ $ id_b   : chr  "X5" "X4" "X1" "X5" ...
+ $ xa     : num  2 14 10 10 2 14 14
+ $ ya     : num  6 10 9 9 6 10 10
+ $ xb     : num  10 4 2 10 4 10 10
+ $ yb     : num  4 10 6 4 10 4 9
+ $ flow_ab: num  1 2 3 3 1 1 4
+ $ flow_ba: num  2 3 2 5 2 1 5
+```
+
+
+### Reshape the flows
+
+The data we have right now in `flows_by_ca` is in a classic long format. That is, we have rows for both the number of trips from A to B and from B to A. `{flowmapper}` requires the data to be in different format, where there is only one row for each pair of `id_a` and `id_b` and two columns, one for the flow from `id_a` to `id_b` and one for the flow from `id_b` to `id_a`.
```

Comment:
same as above, can `{od}` do this with one function instead of my code block below?

Reply:
Let me check... Still relevant?

Reply:
@Robinlovelace Maybe relevant, but not for this vignette, as `{flowmapper}` update simplified the input data format (but you know that already)